tools/lsp/server_loop.rs

```diff
@@ -283,10 +283,23 @@ impl OffsetToPositionMapper {
     }
 }
 
+#[cfg(feature = "preview-api")]
+pub struct PreviewApi {
+    pub highlight: Box<
```

Comment:
Not a blocker, but maybe PreviewApi could be a trait .
The other possibility is just to be `super::PreviewAPI` which would be a type or a module that have its own function.
eg: `super::preview_api::highlight(...)`   and the preview_api is implemented differently in main.rs and in wasm_main.rs
That is currently what `ServerNotifier` is, a type with the same API implemented differently in main.rs and wasm_main.rs 

Anyway, just some ideas, but this is also working.